File: atlantic_report/atlantic_report/report/supplier_do_report___summary/supplier_do_report___summary.py
# Copyright (c) 2025
# Supplier DO Report - Summary

import frappe

def execute(filters=None):
    if not filters:
        filters = {}

    columns = get_columns()
    data = get_data(filters)

    return columns, data


def get_columns():
    return [
        {
            "label": "PR Date",
            "fieldname": "pr_date",
            "fieldtype": "Data",
            "width": 110
        },
        {
            "label": "PR No",
            "fieldname": "pr_no",
            "fieldtype": "Link",
            "options": "Purchase Receipt",
            "width": 140
        },
        {
            "label": "PO Date",
            "fieldname": "po_date",
            "fieldtype": "Data",
            "width": 110
        },
        {
            "label": "Supplier Name",
            "fieldname": "supplier",
            "fieldtype": "Data",
            "width": 200
        },
        {
            "label": "Received By",
            "fieldname": "received_by",
            "fieldtype": "Data",
            "width": 140
        },
        {
            "label": "Status",
            "fieldname": "status",
            "fieldtype": "Data",
            "width": 100
        },
    ]


def get_data(filters):
    sql = """
        SELECT
            DATE_FORMAT(pr.posting_date, '%%d-%%b-%%y') AS pr_date,
            pr.name AS pr_no,
            DATE_FORMAT(po.transaction_date, '%%d-%%b-%%y') AS po_date,
            pr.supplier AS supplier,
            pr.custom_received_by AS received_by,
            pr.status AS status
        FROM
            `tabPurchase Receipt` AS pr
        LEFT JOIN
            `tabPurchase Receipt Item` AS pri ON pri.parent = pr.name
        LEFT JOIN
            `tabPurchase Order` AS po ON po.name = pri.purchase_order
        WHERE
            pr.posting_date BETWEEN %(from_date)s AND %(to_date)s
        GROUP BY
            pr.name, po.name
        ORDER BY
            pr.posting_date ASC, pr.name ASC
    """

    return frappe.db.sql(sql, filters, as_dict=True)

File: atlantic_report/atlantic_report/report/supplier_do_report___summary/supplier_do_report___summary.js
// Supplier DO Report - Summary

frappe.query_reports["Supplier DO Report - Summary"] = {
    filters: [
        {
            fieldname: "from_date",
            label: __("From Date"),
            fieldtype: "Date",
            default: frappe.datetime.month_start(),
            reqd: 1
        },
        {
            fieldname: "to_date",
            label: __("To Date"),
            fieldtype: "Date",
            default: frappe.datetime.get_today(),
            reqd: 1
        },
        {
            fieldname: "supplier",
            label: __("Supplier"),
            fieldtype: "Link",
            options: "Supplier",
            reqd: 0
        }
    ],

    formatter(value, row, column, data, default_formatter) {
        // Highlight PR No as link (default behaviour)
        value = default_formatter(value, row, column, data);

        // Bold Supplier Name column
        if (column.fieldname === "supplier_name") {
            value = `<b>${value}</b>`;
        }

        return value;
    }
};
